docs/src/developing/intro/programs.md

```diff
@@ -0,0 +1,58 @@
+---
+title: What are Solana Programs?
+description: "A Solana Program, aka smart contract, is the executable code that interprets the instructions on the blockchain. There are two types: Native and On-Chain."
+keywords: ""
+---
+
+Solana Programs, often referred to as "_smart contracts_" on other blockchains, are the executable code that interprets the instructions sent inside of each transaction on the blockchain. They can be deployed directly into the core of the network as [Native Programs](#native-programs), or published by anyone as [On Chain Programs](#on-chain-programs). Programs are the core building blocks of the network and handle everything from sending tokens between wallets, to accepting votes of a DAOs, to tracking ownership of NFTs.
+
+Both types of programs run on top of the [Sealevel runtime](https://medium.com/solana-labs/sealevel-parallel-processing-thousands-of-smart-contracts-d814b378192), which is Solana's _parallel processing_ model that helps to enable the high transactions speeds of the blockchain.
+
+## Key points
+
+- Programs are essentially a special type of [Account](../programming-model/accounts.md) that is marked as "_executable_"
+- Programs can own other Accounts
+- Programs can only _change the data_ or _debit_ accounts they own
+- Any program can _read_ or _credit_ another account
+- Programs are considered stateless since the primary data stored in a program account is the its compiled BPF code
+- Programs can be upgraded by their owner (see more on that below)
+
+## Types of programs
+
+The Solana blockchain has two types of programs:
+
+- Native programs
+- On chain programs
+
+### On chain programs
+
+These user programmable programs, often referred to as "_smart contracts_" on other blockchains, are deployed directly to the blockchain for anyone to interact with and execute. Hence the name "on chain"!
+
+In effect, "on chain programs" are any program that is not baked directly into the Solana cluster's core code (like the native programs discussed below).
+
+All these programs are collectively known as the [Solana Program Library](https://spl.solana.com/). And even though the Solana foundation and it's core engineers maintain a few of the on chain programs, any one can create or publish one. On chain programs can also be updated directly on the blockchain by the respective program's Account owner.
+
+### Native programs
+
+Native programs are a special handful of programs that are built directly into the core of the Solana blockchain.
+They are essential in the operation of validator nodes and clusters. These special programs
```

Comment:
The native programs are not essential for the runtime. They are just core capabilities that are treated slightly different that on-chain programs